pkg/clients/ec2/routetable.go

```diff
@@ -143,46 +158,67 @@ func LateInitializeRT(in *v1alpha4.RouteTableParameters, rt *ec2.RouteTable) { /
 	}
 }
 
-// CreateRTPatch creates a *v1alpha4.RouteTableParameters that has only the changed
-// values between the target *v1alpha4.RouteTableParameters and the current
+// CreateRTPatch creates a *v1alpha5.RouteTableParameters that has only the changed
+// values between the target *v1alpha5.RouteTableParameters and the current
 // *ec2.RouteTable
-func CreateRTPatch(in ec2.RouteTable, target v1alpha4.RouteTableParameters) (*v1alpha4.RouteTableParameters, error) {
-	currentParams := &v1alpha4.RouteTableParameters{}
+func CreateRTPatch(in ec2.RouteTable, target v1alpha5.RouteTableParameters) (*v1alpha5.RouteTableParameters, error) {
+	currentParams := &v1alpha5.RouteTableParameters{}
 
 	v1beta1.SortTags(target.Tags, in.Tags)
 
 	// Add the default route for fair comparison.
 	for _, val := range in.Routes {
-		if *val.GatewayId == LocalGatewayID {
-			target.Routes = append([]v1alpha4.Route{{
+		if val.GatewayId != nil && *val.GatewayId == LocalGatewayID {
+			target.Routes = append([]v1alpha5.Route{{
 				GatewayID:            val.GatewayId,
 				DestinationCIDRBlock: val.DestinationCidrBlock,
 			}}, target.Routes...)
 		}
 	}
+	SortRoutes(target.Routes, in.Routes)
 
 	LateInitializeRT(currentParams, &in)
 
+	for i := 0; i < len(target.Routes); i++ {
+		target.Routes[i].ClearRefSelectors()
+	}
+	for i := 0; i < len(target.Associations); i++ {
+		target.Associations[i].ClearRefSelectors()
+	}
```

Comment:
Yeah intention is to clear out the refs/selectors which is also why I don't use the range loops above as want to mutate in place for the patch compare. I can go ahead and swap in a copy just in case

Reply:
As near as I could tell since they are nested under Associations they weren't being ignored and couldn't find a way to set the options to ignore them so just cleared them out.